### PR TITLE
#231 - Added ResponseMetricsSlice

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #231 
Added `ResponseMetricsSlice` that wraps a `Slice` and reports responses produced by original `Slice` to `Metrics`

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#269](https://github.com/artipie/artipie/issues/269) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/f81c084c-b4d0-41f6-b54c-d17a01eed3aa); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/ResponseMetricsSliceTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public void shouldReportSuccessResponse() {
+        this.send(RqMethod.GET, new RsWithStatus(RsStatus.OK));
+        MatcherAssert.assertThat(
+            this.metrics.counter("get.success").value(),
+            new IsEqual<>(1L)
+        );
+    }
+
+    @Test
+    public void shouldReportErrorResponse() {
```
Comment (Vatavuk) [inline src/test/java/com/artipie/ResponseMetricsSliceTest.java:67]: @olegmoz add one test for 4xx status code

**Comment by olegmoz** [comment]
@Vatavuk thanks, made the changes by your comments

File: src/main/java/com/artipie/ResponseMetricsSlice.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+ * Returned response reported to metrics as success or error if it's status code is 4xx or 5xx.
+ *
+ * @since 0.9
+ * @todo #231:30min Test request and response forwarding in `ResponseMetricsSlice`.
+ * Test that request coming to `ResponseMetricsSlice` are forwarded as-is to origin slice,
+ * response generated by origin slice is sent to connection as-is.
+ * @todo #231:30min Report exceptions as errors in `ResponseMetricsSlice`.
+ * In case of exceptions in origin slice handling or sending to `Connection` nothing is reported
+ * to `Metrics`. It needs to be fixed, so if origin slice fails or we failed to send response
+ * then error is sent to `Metrics`.
```
Comment (g4s8) [inline src/main/java/com/artipie/ResponseMetricsSlice.java:48]: @olegmoz puzzles are grouped by indentation, like python code. It should be:
```
@todo #213:30min Puzzle1
 Puzzle1 body line 1
 Puzzle1 body line 2
@todo #231:30min Puzzle2
 Puzzle2 body line 1
 Puzzle2 body line 2
```

Comment (olegmoz) [reply src/main/java/com/artipie/ResponseMetricsSlice.java]: @g4s8 fixed it, thanks

**Comment by 0crat** [comment]
Job was finished in 3 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by Vatavuk** [comment]
@sereshqua ping

**Pull request merged**